src/v2/client/startLoginFlow.js

```diff
@@ -22,24 +23,47 @@ const handleProxyIdxResponse = async (settings) => {
   });
 };
 
-export async function startLoginFlow(settings, appState) {
+export async function startLoginFlow(settings) {
   // Return a preset response
   if (settings.get('proxyIdxResponse')) {
     return handleProxyIdxResponse(settings);
   }
 
-  // Use stateToken
+  if (settings.get('overrideExistingToken')) {
+    sessionStorageHelper.removeStateHandle();
+  }
+
+  // Use stateToken from session storage if exists
+  // See more details at ./docs/use-session-token-prior-to-settings.png
+  const stateHandleFromSession = sessionStorageHelper.getStateHandle();
+  if (stateHandleFromSession) {
+    return introspect(settings, stateHandleFromSession)
+      .then((idxResp) => {
+        // 1. abundant the settings.stateHandle given session.stateHandle is still valid
+        settings.set('stateToken', stateHandleFromSession);
+        // 2. chain the idxResp to next handler
+        return idxResp;
+      })
+      .catch(() => {
+        // 1. remove session.stateHandle
+        sessionStorageHelper.removeStateHandle();
+        // 2. start the login again in order to introspect on settings.stateHandle
+        return startLoginFlow(settings);
+      });
+  }
+
+  // Use stateToken from options
   const stateHandle = settings.get('stateToken');
   if (stateHandle) {
-    return introspect(settings, appState);
+    return introspect(settings, stateHandle);
```

Comment:
we should store this `stateHandle` in session storage right ?

Reply:
Yes, it's handled at success callback in BaseLoginRouter

Reply:
See the image I left to `docs` folder.